src/api/__tests__/Item.test.js

```diff
@@ -359,6 +371,97 @@ describe('api/Item', () => {
                 expect(item.errorCallback).toBe(noop);
             });
         });
+
+        describe('with a cached item', () => {
+            beforeEach(() => {
+                item.getCache = () => ({
+                    get: jest.fn().mockReturnValue('success'),
+                    has: jest.fn().mockReturnValue(true),
+                    merge: jest.fn(),
+                    set: jest.fn(),
+                });
+                item.shareSuccessHandler = jest.fn();
+                item.errorHandler = jest.fn();
+                item.getUrl = jest.fn().mockReturnValue('url');
+                item.xhr = {
+                    put: jest.fn().mockReturnValueOnce(Promise.resolve('success')),
+                };
+            });
+
+            afterEach(() => {
+                jest.resetModules();
+                jest.restoreAllMocks();
+            });
+
+            test('should skip the xhr if the item is cached', async () => {
+                const successCallback = jest.fn();
+                const errorCallback = jest.fn();
+                await item.share(file, 'access', successCallback, errorCallback);
+                expect(successCallback).toHaveBeenCalledWith('success');
+                expect(errorCallback).not.toHaveBeenCalled();
+                expect(item.errorHandler).not.toHaveBeenCalled();
+                expect(item.shareSuccessHandler).not.toHaveBeenCalled();
+                expect(item.xhr.put).not.toHaveBeenCalled();
+                expect(item.getUrl).not.toHaveBeenCalled();
+                expect(item.successCallback).toBeUndefined();
+                expect(item.errorCallback).toBeUndefined();
+                expect(item.id).toBeUndefined();
+            });
+
+            test('should make an xhr if the item is cached, but options.forceFetch is true', async () => {
+                await item.share(file, 'access', 'success', 'error', {
+                    forceFetch: true,
+                });
+                expect(item.shareSuccessHandler).toHaveBeenCalledWith('success');
+                expect(item.errorHandler).not.toHaveBeenCalled();
+                expect(item.successCallback).toBe('success');
+                expect(item.errorCallback).toBe('error');
+                expect(item.id).toBe('id');
+                expect(item.xhr.put).toHaveBeenCalledWith({
+                    url: 'url',
+                    data: { shared_link: { access: 'access' } },
+                });
+                expect(item.getUrl).toHaveBeenCalledWith('id');
+                expect(item.getCacheKey).toHaveBeenCalledWith('id');
+            });
+        });
+
+        describe('with additional data', () => {
+            beforeEach(() => {
+                item.shareSuccessHandler = jest.fn();
```

Comment:
I recommend using `jest.spyOn` rather than direct assignment when possible. Migrating to TypeScript will make it more difficult to overwrite properties in this way.